### PR TITLE
Front update proposal

### PR DESCRIPTION
**Pull request by bencharp**
image

image

image

### CODE REVIEW

**Review by DougReeder — APPROVED** [review]
Pretty!  I like that fewer fonts are required.

**Review by raucao — CHANGES_REQUESTED** [review]
Great new direction in my opinion! :clap:  However, I think it's still a bit too rough around the edges to merge immediately. Here are some points I personally think should be addressed (but those are just my opinion, so if everyone else disagrees, it also shouldn't prevent us from merging).

### Logo

Absolutely love the idea of turning it into a stylized lowercase "a" for *a*rmadietto! However, the tiny RS logo in the middle seems unnecessary to me, and I think makes it look worse than if it weren't there. Also, as stated in the other PR before, I think it would be nice to have the logo color be the same yellow as the current Armadietto is using, to recognize it as Armadietto, and to not re-use the RS logo color. I think a yellow Armadietto logo with only the "a" shape would look fantastic next to a remoteStorage logo.

### Colors

Analog to re-using the yellow that currently makes Armadietto easily recognizable, I think the RS orange/red is a bit aggressive, and its use throughout the UI makes it look less friendly than what it currently looks like. I'd propose to experiment with keeping the current color scheme, and only using the orange for the RS logo itself, and perhaps the external links to RS websites/pages.

### remoteStorage protocol blurb

I think the extra remoteStorage blurb with the building block protocol explanations is not useful for end users in their server UI. The explanation underneath should be enough, and could just link the website from the "remoteStorage" word in the text. Also, I think it should only appear on the very front page, and not on the signup page, authorization page, or any other views that aren't the front page.

File: lib/views/auth.html
Comment (raucao) [inline lib/views/auth.html:5]: Shouldn't this be the other way around, because the ID here is actually the URL, while the host is used as the app name?

**Comment by bencharp** [comment]
@raucao, thanks for you good comments.
I'm trying to get this down quickly, but, and just to make sure I've got it right; when you talk about armadietto yellow, you're talking about this one `#f4a903` , the one that is used for links in the current interface? :thinking:

**Comment by raucao** [comment]
> when you talk about armadietto yellow, you're talking about this one `#f4a903` , the one that is used for links in the current interface?

Yes, that one! Kinda cool that GitHub even shows a color preview in the comment for hex colors now. :)

**Comment by bencharp** [comment]
Some screenshots to have an overview.

image

image

image

image

**Comment by raucao** [comment]
Looking fantastic! :clap:

**Comment by raucao** [comment]
I added some commits, mostly addressing trailing whitespace characters.

I removed the footer entirely now, because it's merely repeating the same information that is already contained in the hero paragraph, and I think it looks much cleaner without the superfluous 2 extra paragraphs added.

I also tweaked some styles for small screens. Let me know if you like it...

Before / after:

**Comment by raucao** [comment]
@bencharp Have you properly tested the authorize screen with your branch? Both buttons don't do anything for me in the latest version, unfortunately.

**Comment by bencharp** [comment]
> @bencharp Have you properly tested the authorize screen with your branch? Both buttons don't do anything for me in the latest version, unfortunately.

@raucao Sorry, I forgot to remove the `preventDefault()` from the `togglePassword` in the `auth.html` file
*(I have many gaps in `js`, corrections and feedback are welcome...)*

**Review by raucao — APPROVED** [review]
I think it's good enough to merge now. Could use one more review from someone who hasn't added code to this branch.

**Comment by DougReeder** [comment]
I'm still seeing an issue on 375x667 (iPhone SE) screens:

armadietto-layout

**Comment by DougReeder** [comment]
There's an inconsistency between the inline scripts this adds and our Content Security Policy.  Don't merge yet.

**Review by DougReeder — CHANGES_REQUESTED** [review]
Toggling light and dark mode uses an inline script. Allowing that greatly decreases our security.  I will look at how to do this in a secure way.

**Comment by DougReeder** [comment]
All that's needed now is to move the font to its own file (our Content Security Policy doesn't allow data: URLs) and a final check, and we're ready to merge!

**Review by DougReeder — APPROVED** [review]
The implementation and Content Security Policy are now aligned.

The font should be moved to its own file, and it would be vary nice to have the sprites work on Chrome MacOS.

**Comment by bencharp** [comment]
> The implementation and Content Security Policy are now aligned.
> 
> The font should be moved to its own file, and it would be vary nice to have the sprites work on Chrome MacOS.

@DougReeder: do you have more info on the sprite svg problem on Chrome MacOS?

I moved the font to its own file, but I'm having a bit of trouble understanding the points made above (the inline javascript in the views and the base64 link for the font file...) and I would have liked to understand these issues a bit better :thinking:

**Comment by DougReeder** [comment]
Mozilla has a good article on the *what* (https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP), but the *why* boils down to this: browsers support a lot of features, some of which were designed in an environment with few threats, and turned out to be easy to hack. A CSP tells the browser what features your page does and doesn't use. Malware that corrupts the page can't do anything that the CSP forbids. So what we don't *need*, we disable. Inline JavaScript is relatively easy to trick the browser into creating, and JavaScript in files can do everything we need. So we disable inline JavaScript, and serve our JavaScript from files (which must come from the same host as the page).  Data URLs are likewise insecure, and we can serve the font as a file.  https://csp.withgoogle.com/docs/why-csp.html  has more.

With this PR, we're actually loosening the CSP - before, we weren't using JavaScript or fonts at all, so they were disabled.

**Comment by DougReeder** [comment]
On FF 96.0.3 on MacOS 11.6.2, the sprites appear as expected.  
image
On Chrome 98.0.4758.66 they don't appear at all, and there's no console message.

**Comment by bencharp** [comment]
I have an error with loading the font since we moved it to its own file. : 
> downloadable font: download failed (font-family: "Outfit" style:normal weight:100..900 stretch:100 src index:0): status=2152398850 source: https://store.local/assets/outfit-variablefont_wght.woff2

I also have a warning about the Content Security Policy :
> Content Security Policy: This site (https://...) has a Report-Only policy with no reporting URI. CSP will not block this policy and will not be able to report its violations.

**Comment by DougReeder** [comment]
Is it possible you have a proxy or reverse proxy in front of Armadietto that's adding a Content-Security-Policy-Report-Only header?  Can the font file be downloaded using curl or wget?

**Comment by bencharp** [comment]
> Is it possible you have a proxy or reverse proxy in front of Armadietto that's adding a Content-Security-Policy-Report-Only header? 

I am testing Armadietto with Yunohost as part of the development and maintenance of the [Armadietto_ynh package](https://github.com/YunoHost-Apps/armadietto_ynh). Indeed, Yunohost adds a "Content-Security-Policy-Report-Only" header in its nginx conf: https://github.com/YunoHost/yunohost/blob/dev/conf/nginx/security.conf.inc.

Is there a way to correct this warning either from the Armadietto or Yunohost package?


> Can the font file be downloaded using curl or wget?

Yes, I can download the font file with curl... :thinking:

**Comment by DougReeder** [comment]
A report-uri (and probably a report-to) directive, pointing to a host that accepts violation reports for your domain, must be added to the Content-Security-Policy-Report-Only header, or else it is useless:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/report-uri

Try removing the  Content-Security-Policy directives; Armadietto now sets that itself.  (Setting  default-src data: removes the key protection of a CSP.)  The other header directives should be fine.

Also, try different browsers, and see if they have more useful error messages.

**Comment by raucao** [comment]
> Try removing the Content-Security-Policy directives

It looks like that's a shared file for all Yunohost apps, in which case removing the header would probably be controversial for an upstream merge. Maybe we can adjust the layout on the Armadietto side to work with this somehow?

**Comment by bencharp** [comment]
I just fixed the woff2 font file that turned out to be faulty _(fontsquirrel has trouble converting variable fonts...)_

I propose to leave out the two issues I raised above as they are beyond the scope of the original PR request; They are more issues with the integration of armadietto as a package for yunohost than with the proposed front update...

I'll open a question about this when I could realy test this update in my yunohost package _(I have some doubts about my current conf...)_ :confused: 

I suggest that we re-launch the revisions requests

**Comment by raucao** [comment]
> I just fixed the woff2 font file that turned out to be faulty _(fontsquirrel has trouble converting variable fonts...)_

Great! :clap: 

> I suggest that we re-launch the revisions requests

That means you think it's ready to merge now, and that we should get a final review from a couple of contributors, correct?

**Comment by bencharp** [comment]
> That means you think it's ready to merge now, and that we should get a final review from a couple of contributors, correct?

Correct ! :slightly_smiling_face:

**Comment by DougReeder** [comment]
With three sets of eyes, I think this has been thoroughly reviewed.  If no one objects, I'll merge this Thursday.

**Pull request merged**
